File: src/main/java/leets/leenk/domain/leenk/domain/service/LeenkUpdateService.java
package leets.leenk.domain.leenk.domain.service;

import java.time.LocalDateTime;
import leets.leenk.domain.leenk.domain.entity.Leenk;
import leets.leenk.domain.leenk.domain.entity.Location;
import leets.leenk.domain.media.domain.entity.Media;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class LeenkUpdateService {

    public void updateTitle(Leenk leenk, String title) {
        if (title != null && !title.isBlank()) {
            leenk.updateTitle(title);
        }
    }

    public void updateContent(Leenk leenk, String content) {
        if (content != null && !content.isBlank()) {
            leenk.updateContent(content);
        }
    }

    public void updateStartTime(Leenk leenk, LocalDateTime startTime) {
        if (startTime != null) {
            leenk.updateStartTime(startTime);
        }
    }

    public void updateMaxParticipants(Leenk leenk, Long maxParticipants) {
        if (maxParticipants != null) {
            leenk.updateMaxParticipants(maxParticipants);
        }
    }

    public void updatePlaceName(Location location, String placeName) {
        if (location != null && placeName != null && !placeName.isBlank()) {
            location.updatePlaceName(placeName);
        }
    }

    public void updateMediaUrl(Media media, String newUrl) {
        if (media != null && newUrl != null && !newUrl.isBlank()) {
            media.updateMediaUrl(newUrl);
        }
    }
}

File: src/main/kotlin/leets/leenk/domain/leenk/domain/service/LeenkUpdateService.kt
package leets.leenk.domain.leenk.domain.service

import leets.leenk.domain.leenk.application.dto.request.LeenkUpdateRequest
import leets.leenk.domain.leenk.application.exception.MaxParticipantsTooLowException
import leets.leenk.domain.leenk.domain.entity.Leenk
import leets.leenk.domain.leenk.domain.entity.Location
import org.springframework.stereotype.Service
import java.time.LocalDateTime

@Service
class LeenkUpdateService {
    fun updateLeenk(leenk: Leenk, location: Location, request: LeenkUpdateRequest) {
        updateTitle(leenk, request.title)
        updateContent(leenk, request.content)
        updateStartTime(leenk, request.startTime)
        updateMaxParticipants(leenk, request.maxParticipants)
        updatePlaceName(location, request.placeName)
    }

    private fun updateTitle(leenk: Leenk, title: String?) {
        if (!title.isNullOrBlank()) {
            leenk.updateTitle(title)
        }
    }

    private fun updateContent(leenk: Leenk, content: String?) {
        if (!content.isNullOrBlank()) {
            leenk.updateContent(content)
        }
    }

    private fun updateStartTime(leenk: Leenk, startTime: LocalDateTime?) {
        if (startTime != null) {
            leenk.updateStartTime(startTime)
        }
    }

    private fun updateMaxParticipants(leenk: Leenk, maxParticipants: Long?) {
        if (maxParticipants == null) {
            return
        }

        val currentCount = leenk.currentParticipants

        if (maxParticipants < currentCount) {
            throw MaxParticipantsTooLowException()
        }

        leenk.updateMaxParticipants(maxParticipants)
    }

    private fun updatePlaceName(location: Location?, placeName: String?) {
        if (location != null && !placeName.isNullOrBlank()) {
            location.updatePlaceName(placeName)
        }
    }
}
